### PR TITLE
Pages and Folders for Functional Tests Menu

### PR DESCRIPTION
**Pull request by patole**

### CODE REVIEW

File: packages/functional-tests/src/menu.ts [hunk 1]
```diff
@@ -127,6 +133,20 @@ export class Menu {
 				if (a.text) { a.text.enabled = true; }
 			}
 		}
+
+		// hide next button if no more pages
+		if((this.currentNode.currentPage+1) * pageSize > this.currentNode.menuItems.length) {
```
Comment (tombuMS) [inline packages/functional-tests/src/menu.ts:138]: >this.currentNode.currentPage+1 [](start = 6, length = 30)

nit: Space around operators

File: packages/functional-tests/src/menu.ts [hunk 2]
```diff
@@ -220,21 +270,76 @@ export class Menu {
...
 		backButton.setBehavior(MRE.ButtonBehavior)
 			.onButton('released', () => {
 				this.back();
 				this.show();
 			});
 
 		this.backActors = [backButton, backLabel];
+	
+		this.nextPage.setBehavior(MRE.ButtonBehavior).onButton('released', () => {
+			if(this.currentNode.menuItems.length > ((this.currentNode.currentPage+1) * pageSize)) {
```
Comment (tombuMS) [inline packages/functional-tests/src/menu.ts:316]: >this.currentNode.currentPage+1 [](start = 44, length = 30)

nit: Space around operators

**Review by tombuMS — APPROVED** [review]
:shipit:

File: packages/functional-tests/src/utils/paginate.ts [hunk 3]
```diff
@@ -6,59 +6,28 @@
...
-		const lastSubpageTestName = subpageNames[subpageNames.length - 1];
-		const subpage = paginate(tests, pageSize, [...subpageNames]);
-		if (subpage.length === 1) {
-			buckets.push(subpage[0]);
-		} else if (subpage.length > 1) {
-			buckets.push({
-				label: uniquePrefix(subpageNames[0], lastBucketsLastTestName) + ' - '
-					+ uniquePrefix(lastSubpageTestName, names[0] || ''),
-				action: subpage
+export function populatePages(tests: FactoryMap, pageSize: number, parentNode: MenuNode = null): MenuNode {
```
Comment (stevenvergenz) [inline packages/functional-tests/src/utils/paginate.ts:15]: Could you add docstrings to the exported interfaces/functions here? Will make it easier to maintain in the future.

File: packages/functional-tests/src/menu.ts (hunk 2 above)
Comment (stevenvergenz) [inline packages/functional-tests/src/menu.ts:316]: >((this.currentNode.currentPage+1) * pageSize) [](start = 42, length = 45)

This expression needs a -1, or else currentPage can == menuItems.length, causing out-of-bounds errors.

**Review by stevenvergenz — APPROVED** [review]
:shipit:

**Pull request merged**